collab2/foraging/toolkit/trace.py

```diff
@@ -1,59 +1,65 @@
+import copy
+from typing import Callable, List, Optional
+
 import numpy as np
 import pandas as pd
 
-from collab.foraging.toolkit.utils import generate_grid
-
+from collab2.foraging.toolkit.utils import dataObject 
+from collab2.foraging.toolkit.filtering import filter_by_distance
+from collab2.foraging.toolkit.decay import _decaying_contribution, _exponential_decay
 
-def rewards_trace(distance, rewards_decay):
-    return np.exp(-rewards_decay * distance)
 
+def _generate_food_predictor(
+    rewards: List[pd.DataFrame],
+    foragers: List[pd.DataFrame],
+    local_windows: List[List[pd.DataFrame]],
+    predictor_name: str,
+    decay_function: Callable = _exponential_decay,
+    **decay_function_kwargs,
+) -> List[List[pd.DataFrame]]:
 
-def rewards_to_trace(
-    rewards,
-    grid_size,
-    num_frames,
-    rewards_decay=0.5,
-    start=None,
-    end=None,
-    time_shift=0,
-    grid=None,
-):
-    if start is None:
-        start = 0
+    num_foragers = len(foragers)
+    num_frames = len(foragers[0])
+    predictor = copy.deepcopy(local_windows)
 
-    if end is None:
-        end = num_frames
+    for f in range(num_foragers):
+        for t in range(num_frames):
+            if predictor[f][t] is not None:
 
-    if grid is None:
-        grid = generate_grid(grid_size)
+                predictor[f][t][predictor_name] = 0
 
-    traces = []
 
-    for t in range(start, end):
-        rewt = rewards[t]
-        trace = grid.copy()
-        trace["trace"] = 0
-        trace["time"] = t + 1
-        trace["trace_standardized"] = 0
+                if len(rewards) > 0:
```

Comment:
I am wondering if there is a better way to organize the `rewards` DF. I guess it is safe to assume that rewards are stationary but can disappear if they are consumed. So instead of having a list of DFs with the same `x`,`y` for all frames, we could just maintain a flattened dataframe of all rewards at any given time.

Reply:
That's what I had in the first place and then revised for uniformity. Will think about revising back

Reply:
ok, now we just need a list of length t of frames with columns x and y, just as before, really